test/libsinsp_e2e/fs.cpp

```diff
@@ -0,0 +1,1672 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+Copyright (C) 2024 The Falco Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#include "event_capture.h"
+#include "subprocess.h"
+#include "sys_call_test.h"
+
+#include <gtest/gtest.h>
+
+#include <fcntl.h>
+#include <libsinsp/event.h>
+#include <libsinsp/sinsp.h>
+#include <libsinsp/sinsp_int.h>
+#include <poll.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string>
+#include <unistd.h>
+
+#include <sys/sendfile.h>
+#include <sys/stat.h>
+#include <sys/syscall.h>
+#include <sys/types.h>
+#include <sys/uio.h>
+
+#include <algorithm>
+#include <cassert>
+#include <iostream>
+#include <list>
+#include <mutex>
+
+#define DATA "AAAAAAAAA"
+
+#define FILENAME "test_tmpfile"
+#define DIRNAME "test_tmpdir"
+#define UNEXISTENT_DIRNAME "/unexistent/pippo"
+
+/////////////////////////////////////////////////////////////////////////////////////
+// creat/unlink
+/////////////////////////////////////////////////////////////////////////////////////
+TEST_F(sys_call_test, fs_creat_ulink)
+{
+	int callnum = 0;
+	char bcwd[1024];
+
+	ASSERT_TRUE(getcwd(bcwd, 1024) != NULL);
+	std::string cwd(bcwd);
+	cwd += "/";
+
+	//
+	// FILTER
+	//
+	event_filter_t filter = [&](sinsp_evt* evt) { return m_tid_filter(evt); };
+
+	//
+	// TEST CODE
+	//
+	run_callback_t test = [&](concurrent_object_handle<sinsp> inspector)
+	{
+		int fd = creat(FILENAME, 0644);
+
+		if (fd < 0)
+		{
+			FAIL();
+		}
+
+		ASSERT_TRUE(write(fd, "ABCD", sizeof("ABCD")) >= 0);
+		close(fd);
+		unlink(FILENAME);
+		unlink(FILENAME);
+	};
+
+	//
+	// OUTPUT VALDATION
+	//
+	captured_event_callback_t callback = [&](const callback_param& param)
+	{
+		sinsp_evt* e = param.m_evt;
+		uint16_t type = e->get_type();
+		std::string name(e->get_name());
+
+		#if defined(__x86_64__)
```

Comment:
I don't really like that we are adding per-architecture cases here... since we are supporting more and more architectures in libs i fear these tests will soon become a real mess.

Reply:
I mean: if we have the need to add per-architecture code, it seems we are not actually working at the `libsinsp` level, since all the arch-dependent code should be tested in driver IMHO.
cc @Andreagit97

Reply:
I don't particularly like my solution but, if we want to stick to the libc wrapper, I think is the only one. Otherwise we can explicitly use the syscall number.

Reply:
I understanrd your point; but i am starting to feel that some of these tests would be better suited for the drivers testing framework :/

Reply:
I think that this is good point for _some of them_: in particular the ones where we don't test anything except the syscall (e.g. `sys_call_test.stat`). For the other ones  setting the dropping mode I'm not super sure.